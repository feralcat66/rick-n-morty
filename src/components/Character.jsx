import React from 'react';
import PropTypes from 'prop-types';


const Character = ({ image, name, status }) => (
  <section className='character'>
    <img src={image} alt={name} />
    <h2>{name}</h2>
    <p>{status}</p>
  </section>
);

Character.propTypes = {
  image: PropTypes.string,
  name: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired
};

export default Character;
