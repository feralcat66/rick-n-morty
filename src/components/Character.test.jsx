  import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('Character component', () => {
  it('renders Character', () => {
    const wrapper = shallow(<Character image="a nice little tree" name="hey" description="Ross" />);
    expect(wrapper).toMatchSnapshot();
  });
});
