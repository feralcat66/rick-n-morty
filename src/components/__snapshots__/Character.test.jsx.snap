// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<section
  className="character"
>
  <img
    alt="hey"
    src="a nice little tree"
  />
  <h3>
    hey
  </h3>
  <p>
    Ross
  </p>
</section>
`;
